scene/resources/theme.cpp

```diff
@@ -196,6 +196,18 @@ bool Theme::is_valid_item_name(const String &p_name) {
 	return true;
 }
 
+String Theme::validate_type_name(const String &p_name) {
+	String type_name = p_name.strip_edges();
+	int len = type_name.length();
+	char32_t *buffer = type_name.ptrw();
```

Comment:
Does it make sense to also optimize `Theme::is_valid_type_name()` and `Theme::is_valid_item_name()`? Or is it only needed because you are modifying the buffer?

Reply:
It's mainly for modifying string content in place. I think those two methods could theoretically be slightly more performant if they accessed pointers directly, since that would eliminate some repetitive boundary checks.